torchsnapshot/io_preparer.py

```diff
@@ -280,6 +282,8 @@ def prepare_read(
 
 @torch.jit.script
 def tensor_copy(dst: torch.Tensor, src: torch.Tensor) -> None:
+    if src.dtype == torch.qint8:
```

Comment:
This would cause issue in certain use cases. For example, users may want to quantize tensors on save and load it for inference. It wouldn't make sense in this case to dequantize and then quantize the tensor.

Curious for your use case, can this be performed on the application side once the quantized tensors are loaded? As a next step to streamline this, we can introduce a custom post process function on the restore side.

Reply:
I think we can do this application-side for now.

Thought:  could we add a `was_quantized` bool to the metadata to determine when to dequantize?